dozer-types/src/json_types.rs

```diff
@@ -1,8 +1,55 @@
+use crate::errors::types::SerializationError;
+use crate::types::{DozerDuration, Field, DATE_FORMAT};
 use chrono::SecondsFormat;
 use ordered_float::OrderedFloat;
-use serde_json::{Map, Value};
+use prost_types::value::Kind;
+use prost_types::{ListValue, Struct, Value as ProstValue};
+use serde::{Deserialize, Serialize};
+use serde_json::{Map, Number, Value};
+use std::collections::BTreeMap;
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
 
-use crate::types::{DozerDuration, Field, DATE_FORMAT};
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, PartialOrd, Ord, Hash)]
+pub enum JsonValue {
+    Null,
+    Bool(bool),
+    Number(OrderedFloat<f64>),
+    String(String),
+    Array(Vec<JsonValue>),
+    Object(BTreeMap<String, JsonValue>),
+}
+
+impl Display for JsonValue {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            JsonValue::Null => f.write_str("NULL"),
+            JsonValue::Bool(v) => f.write_str(&format!("{v}")),
+            JsonValue::Number(v) => f.write_str(&format!("{v}")),
+            JsonValue::String(v) => f.write_str(&v.to_string()),
+            JsonValue::Array(v) => {
+                let list: Vec<String> = v.iter().map(|val| format!("{val}")).collect();
+                let data = &format!("[{}]", list.join(","));
+                f.write_str(data)
+            }
+            JsonValue::Object(v) => {
+                let list: Vec<String> = v.iter().map(|(key, val)| format!("{key}:{val}")).collect();
+                let data = &format!("{{ {} }}", list.join(","));
+                f.write_str(data)
+            }
+        }
+    }
+}
+
+// todo: for serde_json conversion
+impl FromStr for JsonValue {
+    type Err = SerializationError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let object: Value = serde_json::from_str(s).unwrap();
```

Comment:
Don't `unwrap` and return `Err`. The `FromStr::Err` type should be `DeserializationError`.

Reply:
got it